### PR TITLE
feat(llama.cpp): enable Flash Attention by default

### PR DESCRIPTION
**Pull request by ttupper92618**
## What

The llama.cpp runner built `Llama(...)` without `flash_attn`, leaving Flash Attention off. This hit gemma especially hard: its interleaved sliding-window attention has per-layer-varying V embeddings, so without FA llama.cpp logs `padding V cache to 4096` and falls back to a `full-size SWA cache` — wasting VRAM and slowing attention.

Enable `flash_attn` by default via a node-local knob `_flash_attn_enabled` (env `SKULK_LLAMA_CPP_FLASH_ATTN`, default on; `=0` to disable on a build whose backend lacks FA kernels), matching the existing `_logits_all_enabled` pattern.

## Measured (kite4, Strix Halo Vulkan, gemma-4-31B-it-Q4_K_M)

A/B on the live node:
```
flash_attn=False: 10.7 tok/s   (logs: "padding V cache", "full-size SWA cache")
flash_attn=True:  11.4 tok/s   (FA engaged: no padding warning)
```

So ~7% decode speedup **plus** the VRAM saving from dropping the full-size SWA-cache fallback. (Context: ~11 tok/s is the memory-bandwidth-bound ceiling for a 31B *dense* Q4 model on this iGPU; FA is a refinement, not a transformation — the dense-model speed itself is expected hardware behavior.)

## Tests / docs

`test_flash_attn_enabled` (default-on + opt-out + truthy/falsy parsing). CHANGELOG entry + `SKULK_LLAMA_CPP_FLASH_ATTN` row in architecture-reference. basedpyright 0, ruff clean, nix fmt clean, no em-dashes.

### CODE REVIEW

File: src/skulk/worker/runner/llama_cpp/runner.py [hunk 1]
```diff
@@ -352,6 +352,26 @@ def _logits_all_enabled() -> bool:
...
+    Flash Attention is the modern llama.cpp default and matters most for models
+    whose per-layer V embeddings differ (gemma's interleaved sliding-window
+    attention): without it llama.cpp pads the V cache and falls back to a
+    full-size SWA cache, wasting VRAM and slowing attention. With it the SWA
+    cache sizes correctly and the padding goes away. It is a load-time
+    construction flag (cannot be toggled per request), so it is a node-local
+    knob: default on, set ``SKULK_LLAMA_CPP_FLASH_ATTN=0`` to disable on a
+    backend whose build lacks Flash Attention kernels.
+    """
+    return os.getenv("SKULK_LLAMA_CPP_FLASH_ATTN", "1").strip().lower() in (
```
[the remark this answers is not in the document]
Comment (ttupper92618) [reply src/skulk/worker/runner/llama_cpp/runner.py:367]: Fixed in the latest commit: architecture.md's llama.cpp engine section now documents the Flash Attention default-on behavior and the SKULK_LLAMA_CPP_FLASH_ATTN=0 opt-out, alongside the existing architecture-reference env-var row.
